### ISSUE TITLE
TypeError: Cannot read property 'where' of undefined

### ISSUE THREAD

**Issue by Rajan** [balderdashy/sails-mongo#189]
I'm facing this error only in production. It works fine in dev.
Already tried re-installing mongodb, sails-mongo to latest versions but still fails.

In User.js, it fails in aftercreate.
afterCreate: function(user, cb) {
        console.log('after create: ' + user);

         params = { asker: 'rajan',
         title: 'What are your favorite topics?',
         answerer: user.username,
         askerThumb: 'http://res.cloudinary.com/askable/image/upload/c_thumb,g_face,h_150,w_150/v1408165493/Profiles/jqpsd3c6ycbjrqtyu0hf.jpg'};

        Question.create(params, function (err, question) {
            if (err) {
                console.log(err);
                res.send(500, err);
            } else {

                Question.publishCreate(question.toJSON());
                //return question.toObject();
            }
        });

In error logs:

/var/www/askable/node_modules/sails-mongo/node_modules/mongodb/lib/mongodb/connection/base.js:245
        throw message;      
              ^
TypeError: Cannot read property 'where' of undefined
    at _afterFetchingJunctorRecords (/var/www/askable/node_modules/sails-mongo/node_modules/waterline-cursor/cursor/populateBuffers.js:131:35)
    at /var/www/askable/node_modules/sails-mongo/lib/collection.js:103:5
    at /var/www/askable/node_modules/sails-mongo/node_modules/mongodb/lib/mongodb/cursor.js:163:16
    at commandHandler (/var/www/askable/node_modules/sails-mongo/node_modules/mongodb/lib/mongodb/cursor.js:709:16)
    at /var/www/askable/node_modules/sails-mongo/node_modules/mongodb/lib/mongodb/db.js:1846:9
    at Server.Base._callHandler (/var/www/askable/node_modules/sails-mongo/node_modules/mongodb/lib/mongodb/connection/base.js:445:41)
    at /var/www/askable/node_modules/sails-mongo/node_modules/mongodb/lib/mongodb/connection/server.js:468:18
    at MongoReply.parseBody (/var/www/askable/node_modules/sails-mongo/node_modules/mongodb/lib/mongodb/responses/mongo_reply.js:68:5)
    at null.<anonymous> (/var/www/askable/node_modules/sails-mongo/node_modules/mongodb/lib/mongodb/connection/server.js:426:20)
    at emit (events.js:95:17)

**Comment by sailsbot** [comment]
Thanks for posting, @Rajan.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
